src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.tsx

```diff
@@ -14,84 +14,212 @@
 
-import Checkbox from 'src/shared/components/checkbox/Checkbox';
+import useSelectableGenomesTable from 'src/content/app/new-species-selector/components/selectable-genomes-table/useSelectableGenomesTable';
 
-import type { Species } from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
+import AddSpecies from 'src/content/app/new-species-selector/components/species-search-field/AddSpecies';
+import { SpeciesSearchField } from 'src/content/app/new-species-selector/components/species-search-field/SpeciesSearchField';
+import SpeciesSearchResultsSummary from 'src/content/app/new-species-selector/components/species-search-results-summary/SpeciesSearchResultsSummary';
+import SpeciesSearchResultsTable from 'src/content/app/new-species-selector/components/species-search-results-table/SpeciesSearchResultsTable';
+import GenomesFilterField from 'src/content/app/new-species-selector/components/genomes-filter-field/GenomesFilterField';
+import { CircleLoader } from 'src/shared/components/loader';
+
+import type { SpeciesSearchResponse } from 'src/content/app/new-species-selector/state/species-selector-api-slice/speciesSelectorApiSlice';
+import type { SpeciesSearchMatch } from 'src/content/app/new-species-selector/types/speciesSearchMatch';
 
 import styles from './BlastSpeciesSelector.scss';
 
-const BlastSpeciesSelector = () => {
-  const dispatch = useAppDispatch();
-  const popularSpeciesList = useAppSelector(getPopularSpecies);
-  const selectedSpeciesList = useAppSelector(getSelectedSpeciesList);
-  const selectedGenomeIds = selectedSpeciesList.map(
-    ({ genome_id }) => genome_id
-  );
+type Props = {
+  onSpeciesAdd: (genomes: SpeciesSearchMatch[]) => void;
+  selectedSpecies: Array<{ genome_id: string }>;
+  onClose: () => void;
+};
 
-  useEffect(() => {
-    if (!popularSpeciesList.length) {
-      dispatch(fetchPopularSpecies());
-    }
-  }, [popularSpeciesList.length]);
+// These additional properties enable the limitation of the number of genomes
+// that can be selected.
+type PropsWithMaximumSelectableSpeciesCount = Props & {
+  maxSelectableGenomesCount: number;
+  // selectedGenomesCount: number;
+};
 
-  const onSpeciesSelection = (isChecked: boolean, species: Species) => {
-    if (isChecked) {
-      dispatch(addSelectedSpecies(species));
-    } else {
-      dispatch(removeSelectedSpecies(species.genome_id));
+/**
+ * NOTE
+ * This component is almost an exact copy of GenomeSelectorBySearchQuery; with the following notable exceptions:
+ * - It starts with an empty query. Meanwhile, GenomeSelectorBySearchQuery always starts with a query
+ *   that user has entered on the main screen of Species Selector
+ * - It has a limit on how many genomes can be selected (we aren't allowing more than 25 in BLAST).
+ *   GenomeSelectorBySearchQuery has no such limit.
+ * It is almost certain that this species selector component will reppear for other tools (VEP, etc.).
```

Comment:
You mean `reappear`?

Reply:
Yes, sorry. Looks like the whole line could have just be deleted. I must have forgotten about it somehow.